renderer/test/js_ethereum_provider_browsertest.cc

```diff
@@ -171,3 +171,21 @@ IN_PROC_BROWSER_TEST_F(JSEthereumProviderBrowserTest, IsMetaMaskWritable) {
       "window.ethereum.isMetaMask";
   EXPECT_FALSE(content::EvalJs(main_frame(), overwrite).ExtractBool());
 }
+
+IN_PROC_BROWSER_TEST_F(JSEthereumProviderBrowserTest, NonConfigurable) {
+  brave_wallet::SetDefaultWallet(
+      browser()->profile()->GetPrefs(),
+      brave_wallet::mojom::DefaultWallet::BraveWallet);
+  const GURL url = https_server_.GetURL("/simple.html");
+  NavigateToURLAndWaitForLoadStop(url);
+  std::string overwrite =
+      R"(try {
+         Object.defineProperty(window, 'ethereum', {
+           writable: true,
+         });
+       } catch (e) {}
+       window.ethereum = 42;
+       typeof window.ethereum === 'object'
```

Comment:
Could you also add similar case for window.solana?

Reply:
Rebased in this commit 1fabbe90bf9cd8337e2dc34360fb72f416701a68